drivers/bridge/bridge.go

```diff
@@ -432,6 +432,45 @@ func parseNetworkOptions(option options.Generic) (*networkConfiguration, error)
 	return config, nil
 }
 
+// Returns the non link-local IPv6 subnet for the containers attached to this bridge if found, nil otherwise
+func getV6Network(config *networkConfiguration, i *bridgeInterface) *net.IPNet {
+	if config.FixedCIDRv6 != nil {
+		return config.FixedCIDRv6
+	}
+
+	if i.bridgeIPv6 != nil && i.bridgeIPv6.IP != nil && !i.bridgeIPv6.IP.IsLinkLocalUnicast() {
+		return i.bridgeIPv6
+	}
+
+	return nil
+}
+
+func isolateNetworks(d *driver, thisV4, thisV6 *net.IPNet, enable bool) error {
+	d.Lock()
+	defer d.Unlock()
```

Comment:
am not too comfortable to hold onto to the lock over the code rather than the data.
there are other places in existing bridge driver code, we seem to hold such locks and must be avoided.